### DISCUSSION TITLE
when using jinja template with filters, the filter applies twice !!

### DISCUSSION THREAD

**Discussion by moathOSA** [apache/superset#19292] [Q&A / Help]
![Screenshot 2022-03-21 at 10-03-35 Orders Dashboard](https://user-images.githubusercontent.com/60176683/159217827-9b663270-180f-4add-b4c4-2805a0336078.png)

this is the generated query in the dashboard and as you see there is some filters applied twice like Currency

* i tried every thing like using the filter name instead of the field name
* i tried to make field name diff from the filter name
making alias for the fields name .... 
nothing work ! 

this is the query with filters 

```
SELECT `orders`.`cancelled` AS `Cancelled`,`orders`.`confirmed` AS `Confirmed`, `orders`.`visible` AS `Visible`, `orders`.`created_at` AS `Created_at`, `orders`.`order_no` AS `Order_No`, `orders`.`amount` AS `Amount`, `orders`.`currency` AS `Currency`, `orders`.`normalized_amount_usd` AS `Normalized_Amount_Usd`, `stores__via__branch_id`.`domain` AS `Branch_Domain`, `orders`.`shipment_type` AS `Shipment_Type`,`Stores`.`deal_growth` AS `Deal_Growth` ,`Stores`.`domain` AS `Stores_Domain`, `Stores`.`template` AS `Stores_Template`, `Stores`.`stage` AS `Stores_Stage`, `Stores`.`country` AS `Stores_Country`, `Stores`.`city` AS `Stores_City` 
FROM `orders`
LEFT JOIN `stores` `Stores` ON `orders`.`store_id` = `Stores`.`id` LEFT JOIN `stores` `stores__via__branch_id` ON `orders`.`branch_id` = `stores__via__branch_id`.`id`
        WHERE 'Amount' != 'dumy condition'

        {% if from_dttm != None %}
	          AND orders.Created_at >= '{{ from_dttm }}' AND orders.Created_at < '{{ to_dttm }}'
	      {% endif %}

        {%- if (filter_values('Domain') is defined) and filter_values('Domain') -%}
            AND
            Stores_Domain IN ({{ "'" + "','".join(filter_values('Domain')) + "'" }})
        {%- endif -%}

        {%- if (filter_values('Country') is defined) and filter_values('Country') -%}
            AND
            Stores_Country IN ({{ "'" + "','".join(filter_values('Country')) + "'" }})
        {%- endif -%}

        {%- if (filter_values('Currency') is defined) and filter_values('Currency') -%}
            AND
            Currency IN ({{ "'" + "','".join(filter_values('Currency')) + "'" }})
        {%- endif -%}
 
        {%- if (filter_values('Shipment Type') is defined) and filter_values('Shipment Type') -%}
            AND
            orders.Shipment_Type IN ({{ "'" + "','".join(filter_values('Shipment Type')) + "'" }})
        {%- endif -%}
        
        {%- if (filter_values('Deal Growth') is defined) and filter_values('Deal Growth') -%}
            AND
            Deal_Growth IN ({{ "'" + "','".join(filter_values('Deal Growth')) + "'" }})
        {%- endif -%}        


```

*************
superset version 1.4.1 

i don't see any similar problem to this problem 

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by villebro** [comment]
If you add `remove_filter=True` to the `filter_values` call the filter won't apply twice. See here: https://github.com/apache/superset/blob/dc575080d7e43d40b1734bb8f44fdc291cb95b11/superset/jinja_context.py#L222-L224

We're considering making this the new default behavior in Superset 2.0, but in the 1.x branch this still needs to be done explicitly.

**Comment by moathOSA** [comment]
@villebro  now every thing applied to the outer query !! 
i have a chart with inner query and after appealing the remove_filter=True all filters goes to the outer query

**Comment by moathOSA** [comment]
@villebro  please help here 
using the remove_filter=True make every thing applied to the outer query 
i have a filter must be applied  inside the join and using remove_filter=True not working for me it's make all filters applied to the outer query only

**Comment by srinify** [comment]
Hey there @moathOSA I'm converting this thread into a discussion, as we haven't been able to identify a clear bug just yet. We're focusing Github Issues in the Superset repo primarily around bugs, and occasionally some proposal items.

Just to set expectations, the Superset contributors (like @villebro and I, as well as 30+ others!) receive a lot of questions and we can't answer all of them. Please use Github Discussions or join the Slack (https://superset.apache.org/community) to get help from the community attached to the Superset project. I also recommend reading up on the Community Guidelines! (https://github.com/apache/superset/blob/master/CODE_OF_CONDUCT.md#slack-community-guidelines)

I'm excited to have you be a part of our community! :D

**Comment by moathOSA** [comment]
any help here plz still have the filter twice even  after using remove_filter=True

**Comment by seandavi** [comment]
Looks like this might have been answered in another discussion: https://github.com/apache/superset/issues/19284#issuecomment-1234443208
